iceoryx_dds_gateway/include/ioxdds/internal/gateway/iox2dds.inl

```diff
@@ -0,0 +1,209 @@
+{
+    iox::LogDebug() << "[Iceoryx2DDSGateway] Starting discovery.";
+    m_runDiscoveryLoop.store(true, std::memory_order_relaxed);
+    while (m_runDiscoveryLoop.load(std::memory_order_relaxed))
+    {
+        iox::capro::CaproMessage msg;
+        while (static_cast<gateway_t*>(this)->getCaProMessage(msg))
+        {
+            discover(msg);
+        }
+        std::this_thread::sleep_until(std::chrono::steady_clock::now()
+                                      + std::chrono::milliseconds(DISCOVERY_PERIOD.milliSeconds<int64_t>()));
+    }
+    iox::LogDebug() << "[Iceoryx2DDSGateway] Stopped discovery.";
+}
+
+template <typename gateway_t, typename subscriber_t, typename data_writer_t>
+inline void
+Iceoryx2DDSGateway<gateway_t, subscriber_t, data_writer_t>::discover(const iox::capro::CaproMessage& msg) noexcept
+{
+    iox::LogDebug() << "[Iceoryx2DDSGateway] <CaproMessage> "
+                    << iox::capro::CaproMessageTypeString[static_cast<uint8_t>(msg.m_type)]
+                    << " { Service: " << msg.m_serviceDescription.getServiceIDString()
+                    << ", Instance: " << msg.m_serviceDescription.getInstanceIDString()
+                    << ", Event: " << msg.m_serviceDescription.getEventIDString() << " }";
+
+    if (msg.m_serviceDescription.getServiceIDString() == iox::capro::IdString("Introspection"))
+    {
+        return;
+    }
+    if (msg.m_subType == iox::capro::CaproMessageSubType::SERVICE)
+    {
+        return;
+    }
+
+    switch (msg.m_type)
+    {
+    case iox::capro::CaproMessageType::OFFER:
+    {
+        auto channel = setupChannel(msg.m_serviceDescription);
+        channel.getSubscriber()->subscribe(SUBSCRIBER_CACHE_SIZE);
+        channel.getDataWriter()->connect();
+        break;
+    }
+    case iox::capro::CaproMessageType::STOP_OFFER:
+    {
+        discardChannel(msg.m_serviceDescription);
+        break;
+    }
+    default:
+    {
+        break;
+    }
+    }
+}
+
+template <typename gateway_t, typename subscriber_t, typename data_writer_t>
+inline void Iceoryx2DDSGateway<gateway_t, subscriber_t, data_writer_t>::forwardingLoop() noexcept
+{
+    iox::LogDebug() << "[Iceoryx2DDSGateway] Starting forwarding.";
+    m_runForwardingLoop.store(true, std::memory_order_relaxed);
+    while (m_runForwardingLoop.load(std::memory_order_relaxed))
+    {
+        forward();
+        std::this_thread::sleep_until(std::chrono::steady_clock::now()
+                                      + std::chrono::milliseconds(FORWARDING_PERIOD.milliSeconds<int64_t>()));
+    };
+    iox::LogDebug() << "[Iceoryx2DDSGateway] Stopped forwarding.";
+}
+
+template <typename gateway_t, typename subscriber_t, typename data_writer_t>
+inline void Iceoryx2DDSGateway<gateway_t, subscriber_t, data_writer_t>::forward() noexcept
+{
+    uint64_t index{0};
+
+    auto guardedVector = m_channels.GetScopeGuard();
+    for (auto channel = guardedVector->begin(); channel != guardedVector->end(); channel++)
```

Comment:
Please always use the preincrement operator for iterators:
```++channel```

Reply:
This is already fixed in the next PR.